src/onegov/form/validators.py

```diff
@@ -302,6 +299,39 @@ def __call__(self, form: 'Form', field: 'Field') -> None:
                 errors.append(error)
                 raise ValidationError(error)
 
+    def _parse_form(self, field, enable_indent_check=True):
+        # XXX circular import
+        from onegov.form import parse_form
+
+        return parse_form(field.data,
+                          enable_indent_check=enable_indent_check)()
+
+
+class ValidFilterFormDefinition(ValidFormDefinition):
+    invalid_field_type = _("Invalid field type for field {label}. For filters "
+                           "only 'select' or 'multiple select' fields are "
+                           "allowed.")
+
+    def __call__(self, form: 'Form', field: 'Field') -> None:
+        from onegov.form.fields import MultiCheckboxField
+
+        super().__call__(form, field)
+
+        # limit the definition to MultiCheckboxField, RadioField which can
+        # be used for filter definition
+        parsed_form = self._parse_form(field, enable_indent_check=False)
```

Comment:
This will cause the form to get parsed twice, I would change `ValidFormDefinition.__call__` so it returns either the parsed form or `None` so you can reuse it in your own `__call__` method.

```python
    parsed_form = super().__call__(form, field)
    if parsed_form is None:
        return None
```

Reply:
It now only gets parsed once, but the last error found in the form will be displayed twice. Don't know why as the 'errors' only contains 3 items...
![Screenshot from 2023-09-12 14-40-44](https://github.com/OneGov/onegov-cloud/assets/124258444/90b2a728-a2ea-4fac-80a5-98645b3a6488)